src/lib/core/fiber.c

```diff
@@ -386,6 +389,18 @@ fiber_delete(struct cord *cord, struct fiber *f);
 static void
 cord_add_garbage(struct cord *cord, struct fiber *f);
 
+/* Called right before fiber's execution. */
+static inline void
+fiber_before_call(struct fiber *f)
+{
+	cord()->call_time = clock_monotonic();
```

Comment:
Calling this function on every yield may be unacceptable. Look, we disable `fiber.top` by default to avoid `rdtsc`.

https://github.com/tarantool/tarantool/blob/ed46feb27087247ab5ee47fcf63f28c3ad842099/src/lib/core/fiber.c#L226-L241

https://github.com/tarantool/tarantool/blob/ed46feb27087247ab5ee47fcf63f28c3ad842099/src/lib/core/fiber.c#L112-L128

`clock_monotonic` is heavier than `rdtsc`.

I discussed the issue with @alyapunov. He said that
1. This needs to be benchmarked. He also said that you already ran some tests. Please share the results if so.
2. We can avoid `clock_monotonic` on yield by introducing a "low-res" timer, which would update the clock used for limiting iteration every 100 ms, because we don't need to be precise when it comes to the limit.

Reply:
done